lib/grpc/transport/http2.ex

```diff
@@ -12,6 +12,10 @@ defmodule GRPC.Transport.HTTP2 do
     %{"content-type" => "application/grpc-web-#{codec.name()}"}
   end
 
+  def server_headers(%{codec: GRPC.Codec.JSON = codec}) do
+    %{"content-type" => "application/#{codec.name()}"}
+  end
```

Comment:
I believe we should add an optional `codec.content_type` callback. IIRC @v0idpwn faced some issue where it could have been handy as well. Let's leave this for now, with the following minor change, but leave a to-do as well:

```suggestion
  # TO-DO: refactor when we add a GRPC.Codec.content_type callback
  def server_headers(%{codec: GRPC.Codec.JSON}) do
    %{"content-type" => "application/json"}
  end
```